### PR TITLE
added csp header to helmet

### PR DESCRIPTION
**Pull request by teuron**
Adding Content-Security-Policy Header to contrib

### CODE REVIEW

File: contrib/lib/src/helmet/policy.rs [hunk 1]
```diff
@@ -399,3 +399,203 @@ impl<'a> Into<Header<'static>> for &'a XssFilter {
...
+        }
+
+        Header::new(Csp::NAME, policy_string)
+    }
+}
+
+/// Defaults to ---
+impl Default for Csp {
+    fn default() -> Csp {
+        // FIXME: what default value?
```
[the remark this answers is not in the document]
Comment (teuron) [reply contrib/lib/src/helmet/policy.rs]: A reasonable default could be `Content-Security-Policy: default-src 'none'; frame-ancestors 'none'` as stated in the [Mozilla Guidlines](https://infosec.mozilla.org/guidelines/web_security#csrf-prevention)

File: contrib/lib/src/helmet/policy.rs [hunk 2]
```diff
@@ -399,3 +399,203 @@ impl<'a> Into<Header<'static>> for &'a XssFilter {
...
+        self.into()
+    }
+}
+
+impl Csp {
+    /// Only allow own Uri to load resource
+    pub const SELF: &'static str = "'self'";
+
+    /// Only allow inline resources
+    pub const NONE: &'static str = "'none'";
```
[the remark this answers is not in the document]
Comment (teuron) [reply contrib/lib/src/helmet/policy.rs]: I plan to add those as well, but I wanted to get a review on the first PR

File: contrib/lib/src/helmet/policy.rs [hunk 3]
```diff
@@ -399,3 +399,203 @@ impl<'a> Into<Header<'static>> for &'a XssFilter {
...
+    style_src: Vec<Cow<'static, str>>,
+    connect_src: Vec<Cow<'static, str>>,
+    font_src: Vec<Cow<'static, str>>,
+    frame_src: Vec<Cow<'static, str>>,
+    img_src: Vec<Cow<'static, str>>,
+    manifest_src: Vec<Cow<'static, str>>,
+    media_src: Vec<Cow<'static, str>>,
+    object_src: Vec<Cow<'static, str>>,
+    script_src: Vec<Cow<'static, str>>,
+}
```
[the remark this answers is not in the document]
Comment (teuron) [reply contrib/lib/src/helmet/policy.rs]: I will add those in a week or two.

**Comment by SergioBenitez** [comment]
I'm not in favor of adding this to `contrib`. CSP is complicated, and I fear that by including it in `helmet`, we'll make it too easy to misconfigure CSP, giving users a false sense of security. Instead, I'd advocate for releasing a more type-safe version of this idea as its own crate, @teuron.

Thank you for brining this up, however!

**Comment by teuron** [comment]
@SergioBenitez I totally understand your opinion, but what shall we do with #264? Shall we close that issue to or mention that this should be done in a separate crate?

**Pull request closed without merging**